src/Generator/Generators/CLI/CLIHeadersTemplate.cs

```diff
@@ -137,6 +137,9 @@ public void GenerateForwardRefs()
             var typeReferences = typeReferenceCollector.TypeReferences;
             var @namespace = ConvertForwardReferencesToNamespaces(typeReferences);
 
+            foreach (var cls in @namespace.Classes)
+                cls.IsStruct = ExtensionMethods.CheckClassIsStructible(cls, Driver);
```

Comment:
We should not be modifying the AST in the text generators. This should be done in a pass.